encoding/marshaller.go

```diff
@@ -0,0 +1,79 @@
+package encoding
+
+import (
+	"bytes"
+	"io"
+
+	cbor "github.com/polydawn/refmt/cbor"
+	"github.com/polydawn/refmt/obj/atlas"
+)
+
+type proxyWriter struct {
+	w io.Writer
+}
+
+func (w *proxyWriter) Write(b []byte) (int, error) {
+	return w.w.Write(b)
+}
+
+// Marshaller is a reusbale CBOR marshaller.
+type Marshaller struct {
+	marshal *cbor.Marshaller
+	writer  proxyWriter
```

Comment:
I don't understand why this writer would want to be in a stateful field?

Reply:
I need to be able to change the marshaller's writer after creating it so I'm updating the `w` field on this "proxy" writer.

Reply:
... ok, so that actually also fits with "things that get convoluted because refmt constructors bundled a bunch of things together that maybe shouldn't be".  hrmph.  I think I have some work to do down there because this just shouldn't be this complicated up here for the caller.

Reply:
K, I'm going to mark this resolved... I'm sold on merging "the thing we need now".

Over this last week, I worked on some more benchmarking and potential refactors to refmt, but they all ballooned in scope rapidly.  While there's definitely stuff to be done there, I need to figure out how to make those more bite-sized, and let's not wait on it.  (Also, since benchmarking this again, it seems to already be a strict improvement on the existing stuff.  So that's a powerful argument to just moving forward.  More on that in the main comment thread in a moment.)